app/assets/javascripts/code/services/_sensors.js

```diff
@@ -93,38 +93,39 @@ export const sensors = (params, $http) => {
       if (sensor) {
         return _(this.availableParameters).find(function(parameter) { return (parameter.id == sensor["measurement_type"]) });
       } else {
-        return null;
+        return ALL_PARAMETER;
       }
     },
     onSelectedParameterChange: function(selectedParameter, oldValue) {
       console.log('onSelectedParameterChange() - ', selectedParameter)
       if (selectedParameter === oldValue) return; // first angular watch run
       params.update({selectedSessionIds: []});
-      if (hasChangedToAll(selectedParameter)) {
-        this.availableSensors = sort(Object.values(this.sensors));
-        params.update({data: {sensorId: ""}});
+      if (selectedParameter.id === ALL_PARAMETER.id) {
+        this.availableSensors = [ALL_SENSOR].concat(sort(Object.values(this.sensors)));
+        params.update({data: {sensorId: ALL_SENSOR.id}});
       } else {
-        this.availableSensors = sort(Object.values(this.sensors).filter(sensor => sensor.measurement_type === selectedParameter.id));
+        this.availableSensors = [ALL_SENSOR].concat(sort(Object.values(this.sensors).filter(sensor => sensor.measurement_type === selectedParameter.id)));
         const sensorId = defaultSensorIdForParameter(selectedParameter, this.availableSensors);
         params.update({ data: { sensorId }});
       }
     },
+    // changed params.get('data').sensorId
     onSelectedSensorChange: function(newSensorId) {
       console.log('onSelectedSensorChange() - ', newSensorId);
-      if(hasChangedToAll(newSensorId)){
-        params.update({data: {sensorId: ""}});
-      }
+      if (!newSensorId) return;
+      if(newSensorId === ALL_SENSOR.id) params.update({data: {sensorId: ALL_SENSOR.id}});
       var sensor = this.findSensorById(newSensorId);
       var parameterForSensor = this.findParameterForSensor(sensor);
       this.selectedParameter = parameterForSensor;
     },
     buildSensorId: function(sensor) {
       return buildSensorId(sensor);
     },
+    // dropdown
     onSensorsSelectedIdChange: function(newValue, oldValue, callback) {
       console.log("onSensorsSelectedIdChange - ", newValue, " - ", oldValue);
 
-      if(hasChangedToAll(newValue)) return;
+      if(!newValue || newValue === ALL_SENSOR.id) return;
```

Comment:
is it still possible that `!newValue` is true? Since we have ALL_SENSOR default now? I don't know, just asking :)

Reply:
Changed this to make it clearer